lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1089 @@
+    switch (TI.getSimpleMode()) {
+    case SimpleTypeMode::NearPointer:
+    case SimpleTypeMode::FarPointer:
+    case SimpleTypeMode::HugePointer:
+    case SimpleTypeMode::NearPointer32:
+    case SimpleTypeMode::FarPointer32:
+    case SimpleTypeMode::NearPointer64:
+    case SimpleTypeMode::NearPointer128:
+      return model::PrimitiveTypeKind::PointerOrNumber;
+    default:
+      return model::PrimitiveTypeKind::Invalid;
+    }
+  }
+
+  switch (TI.getSimpleKind()) {
+  case SimpleTypeKind::Boolean8:
+  case SimpleTypeKind::Boolean16:
+  case SimpleTypeKind::Boolean32:
+  case SimpleTypeKind::Boolean64:
+  case SimpleTypeKind::Boolean128:
+  case SimpleTypeKind::Byte:
+  case SimpleTypeKind::UInt16:
+  case SimpleTypeKind::UInt32:
+  case SimpleTypeKind::UInt64:
+  case SimpleTypeKind::UnsignedCharacter:
+  case SimpleTypeKind::UInt16Short:
+  case SimpleTypeKind::UInt32Long:
+  case SimpleTypeKind::UInt64Quad:
+  case SimpleTypeKind::UInt128Oct:
+  case SimpleTypeKind::UInt128:
+    return model::PrimitiveTypeKind::Unsigned;
+  case SimpleTypeKind::SignedCharacter:
+  case SimpleTypeKind::WideCharacter:
+  case SimpleTypeKind::NarrowCharacter:
+  case SimpleTypeKind::Character16:
+  case SimpleTypeKind::Character32:
+  case SimpleTypeKind::SByte:
+  case SimpleTypeKind::Int32Long:
+  case SimpleTypeKind::Int32:
+  case SimpleTypeKind::Int64Quad:
+  case SimpleTypeKind::Int64:
+  case SimpleTypeKind::Int128Oct:
+  case SimpleTypeKind::Int128:
+    return model::PrimitiveTypeKind::Signed;
+  case SimpleTypeKind::Float16:
+  case SimpleTypeKind::Float32:
+  case SimpleTypeKind::Float64:
+  case SimpleTypeKind::Float80:
+  case SimpleTypeKind::Float128:
+    return model::PrimitiveTypeKind::Float;
+  default:
+    return model::PrimitiveTypeKind::Invalid;
+  }
+}
+
+void PDBImporterTypeVisitor::createPrimitiveType(TypeIndex TI) {
+  using namespace model;
+  PrimitiveTypeKind::Values Kind = codeviewSimpleTypeEncodingToModel(TI);
+
+  if (TI.getSimpleKind() == SimpleTypeKind::Void) {
+    auto VoidType = Model->getPrimitiveType(PrimitiveTypeKind::Void, 0);
+
+    if (TI.getSimpleMode() == SimpleTypeMode::Direct)
+      ProcessedTypes[TI] = VoidType;
+    else {
+      // Pointer.
+      auto TypedefTy = makeType<TypedefType>();
+      auto TypeDef = cast<TypedefType>(TypedefTy.get());
+      std::vector<Qualifier> Qualifiers;
+      if (TI.getSimpleMode() == SimpleTypeMode::NearPointer32) {
+        constexpr uint64_t PointerSize = 4;
+        Qualifiers.push_back({ Qualifier::createPointer(PointerSize) });
+      } else if (TI.getSimpleMode() == SimpleTypeMode::NearPointer64) {
+        constexpr uint64_t PointerSize = 8;
+        Qualifiers.push_back({ Qualifier::createPointer(PointerSize) });
+      }
+
+      QualifiedType Qty(VoidType, Qualifiers);
+      TypeDef->UnderlyingType = Qty;
+      auto TyPath = Model->recordNewType(std::move(TypedefTy));
+      ProcessedTypes[TI] = TyPath;
+    }
+    return;
```

Comment:
I'd wrap the rest of the code in an `else` instead of early `return`ing. Same below.